### PR TITLE
Added EMC to SkyStone and CertusQuartz

### PR DESCRIPTION
**Pull request by FireBall1725**

### CODE REVIEW

**Comment by FireBall1725** [comment]
Don't pull in yet.

**Comment by yueh** [comment]
You should bug Pahimar about a real api jar, so we do not have to pull all that useless stuff into indev.

**Comment by TheJulianJES** [comment]
@FireBall1725 Why has SkyStone a low EMC value? It would be better if it is higher than obsidian because SkyStone is really uncommon I think. (Just my Opinion)

File: src/main/java/appeng/integration/IntegrationType.java [hunk 1]
```diff
@@ -19,8 +19,12 @@
 package appeng.integration;
 
 
+import scala.Int;
```
Comment (TheJulianJES) [inline src/main/java/appeng/integration/IntegrationType.java]: Is this Scala import needed?

**Comment by FireBall1725** [comment]
@yueh meh

**Comment by FireBall1725** [comment]
Hold on merging this pr,
having issues with AE2 detecting EE3
"Equivalent Exchange 3 - Integration Disabled"

**Comment by FireBall1725** [comment]
@TheJulianJES probably, maybe ill just add it to a config somewhere so you can just set it to what ever you want...

**Comment by TheJulianJES** [comment]
@FireBall1725 Yeah, I would like that :)

**Comment by FireBall1725** [comment]
Right now trying to figure out why it doens't detect EE3

**Comment by FireBall1725** [comment]
@TheJulianJES found out that there is no need for a config in AE2, EE3 creates the config file.  We just set the defaults, and if you want to configure any of the default EMC for any of the mods just edit the one file.

**Comment by TheJulianJES** [comment]
@Fireball1725 Oh, that's nice :)

**Comment by FireBall1725** [comment]
This is good to go, the only thing left is to figure out if we want to add EMC to other things, thinking maybe yes...

**Comment by FireBall1725** [comment]
@TheJulianJES and the scala import is on error as well, just noticed your comment about the import.

**Comment by TheJulianJES** [comment]
@FireBall1725 Okay :)

**Comment by yueh** [comment]
`IIntegrationModule.preInit()` is completely unnecessary. Both `IntegrationStage.PRE_INIT` and `IntegrationStage.INIT` is called during `FMLInitializationEvent` and `PRE_INIT` is really only being used to setup the integration itself. This also means if some integrations are depending on eachother. This also means when they are not fully initialized at this point, it might fail.

The EE3 API sadly does not state when `EnergyValueRegistryProxy.addPreAssignedEnergyValue()` needs to be called (PreInit/Init). If it needs to be called during PreInit, then it is not possible with our current integration system.
But I currently cannot say, if we could move it to PreInit without causing issues.

I also tried a few things with it, but for whatever reasons I could not convince it to assign an EMC value for charged certus nor add any recipe via `RecipeRegistryProxy.addRecipe()`. For example charged certus, the cutting knifes or dusts.

Btw. there is also `certusQuartzCrystal().maybeStack(size)`, so you do not have to tinker with `new ItemStack` directly and things like the metadata.

**Comment by FireBall1725** [comment]
Im going to close this in favor of a different design of adding EMC, will reopen a new PR once that is completed.

**Pull request closed without merging**